force-app/main/default/classes/AFFL_MultiRecordTypeMapper.cls

```diff
@@ -333,29 +316,30 @@ public virtual with sharing class AFFL_MultiRecordTypeMapper {
      * @param affl The affiliation to check for program enrollment creations
      * @return Whether or not the affiliation requires a program enrollment to be auto-created.
      ********************************************************************************************************/
-    public virtual Boolean needsProgramEnrollment(Affiliation__c affl) {
```

Comment:
I am a bit worried here since we are changing the logic entirely here. It looked like we were passing affiliation and now affiliation mappings. Just want to double check if this is intended and has no risks. Maybe it was not designed this way but might already became a feature atm.

Reply:
This method is still passing in an Affiliation record and acting on it based on abstractions of Affiliation Mappings as configured in the constructor.

I retrofitted this a touch. Originally three maps were made from the Primary Affiliation Settings of the raw account record type value to the three auto enrollment settings. This map used the raw Account Record Type field in the method, meaning it wouldn't work on API names as it was always compared to Affiliation__c.Affiliation_Type__c, which resolves to the. Affiliation's Account's RecordType's Name (a label).

The solve implemented was to instead initialize a map of Primary Affiliation Mappings by their Account Record Type Name (resolving from Developer Name to Name and discarded invalid. mappings in the process). That map is then used in this method to pull its related and relevant fields by using map.get(Affiliation__c.Affiliation_Type__c).

Effectively I ended up trading three inadequate abstractions for one adequate map.